ecs_deploy/cli.py

```diff
@@ -28,53 +28,96 @@ def get_client(access_key_id, secret_access_key, region, profile):
+@click.option('--diff/--no-diff', default=True,
+              help='Print which values were changed in the task definition (default: --diff)')
+@click.option('--deregister/--no-deregister', default=True,
+              help='Deregister or keep the old task definition (default: --deregister)')
+@click.option('--rollback/--no-rollback', default=False,
+              help='Rollback to previous revision, if deployment failed (default: --no-rollback)')
+@click.option('--exclusive-env', is_flag=True, default=False,
+              help='Set the given environment variables exclusively and remove all other pre-existing env variables from all containers')
+@click.option('--exclusive-secrets', is_flag=True, default=False,
+              help='Set the given secrets exclusively and remove all other pre-existing secrets from all containers')
+@click.option('--sleep-time', default=1, type=int,
+              help='Amount of seconds to wait between each check of the service (default: 1)')
+@click.option('--slack-url', required=False,
+              help='Webhook URL of the Slack integration. Can also be defined via environment variable SLACK_URL')
+@click.option('--slack-service-match', default=".*", required=False,
+              help='A regular expression for defining, which services should be notified. (default: .* =all). Can also be defined via environment variable SLACK_SERVICE_MATCH')
+@click.option('--exclusive-ulimits', is_flag=True, default=False,
+              help='Set the given ulimits exclusively and remove all other pre-existing ulimits from all containers')
+@click.option('--exclusive-system-controls', is_flag=True, default=False,
+              help='Set the given system controls exclusively and remove all other pre-existing system controls from all containers')
+@click.option('--exclusive-ports', is_flag=True, default=False,
+              help='Set the given port mappings exclusively and remove all other pre-existing port mappings from all containers')
+@click.option('--exclusive-mounts', is_flag=True, default=False,
+              help='Set the given mount points exclusively and remove all other pre-existing mount points from all containers')
+@click.option('--volume', type=(str, str), multiple=True, required=False,
+              help='Set volume mapping from host to container in the task definition.')
+@click.option('--add-container', type=str, multiple=True, required=False,
+              help='Add a placeholder container in the task definition.')
+@click.option('--remove-container', type=str, multiple=True, required=False,
+              help='Remove a container from the task definition.')
+def deploy(cluster, service, tag, image, command, health_check, cpu, memory, memoryreservation, privileged, essential,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using more than five non-keyword parameters in a function definition can indicate that the function is difficult to understand and maintain. Try to combine parameters in objects or refactor the function into small functions.